### PR TITLE
zephyr: allow the serial recovery without any GPIO defined

### PR DESCRIPTION
**Pull request by nvlsianpu**
Despite MCUBOOT support waiting on serial interface it always
requires GPIO pin which is a mistake.

Introduced intermediate CONFIG_MCUBOOT_SERIAL_USE_GPIO Kconfig
hidden option and used it for make code related to the pin-detection
excluded when CONFIG_BOOT_SERIAL_WAIT_FOR_DFU=y.

if someone will find out that usage of GPIO trigger and waiting on
DFU make any sens then CONFIG_MCUBOOT_SERIAL_USE_GPIO might be made
configurable. For now I don't see a sense in making that.

### CODE REVIEW

**Comment by nvlsianpu** [comment]
cc @nordicjamie

**Comment by nvlsianpu** [comment]
cc @nordicjm

File: boot/zephyr/Kconfig.serial_recovery [hunk 1]
```diff
@@ -20,6 +19,11 @@ menuconfig MCUBOOT_SERIAL
 
 if MCUBOOT_SERIAL
 
+config MCUBOOT_SERIAL_USE_GPIO
+	bool
+	default y if! BOOT_SERIAL_WAIT_FOR_DFU
```
Comment (nordicjm) [inline boot/zephyr/Kconfig.serial_recovery:24]: Space before !

**Review by nordicjm — COMMENTED** [review]
Typo in commit *make any sense

**Review by d3zd3z — COMMENTED** [review]
I'm happy with the change, with the space fixed.

**Review by d3zd3z — CHANGES_REQUESTED** [review]
The space really does need to be fixed.

**Review by nordicjm — COMMENTED** [review]
Typo in commit still:
`DFU make any sens then CONFIG_MCUBOOT_SERIAL_USE_GPIO might be made`
->
`DFU make any sense then CONFIG_MCUBOOT_SERIAL_USE_GPIO might be made`

**Comment by nvlsianpu** [comment]
@d3zd3z Can you revisit?

**Review by nordicjm — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@d3zd3z Pleas revisit

**Comment by nordicjm** [comment]
Closing as this is no longer needed, there is now a serial recovery entrance selection and GPIO can be disabled. Though this shows an issue with indication as that needs to select GPIO, will raise a new PR for that.

**Pull request closed without merging**